src/Plugin/Patches.php

```diff
@@ -146,12 +123,44 @@ public function activate(Composer $composer, IOInterface $io): void
                 'type' => 'string',
                 'default' => 'patches.json',
             ],
+            'patches-lock-file' => [
+                'type' => 'string',
+                'default' => 'patches.lock.json',
+            ],
             "ignore-dependency-patches" => [
                 'type' => 'list',
                 'default' => [],
             ],
         ];
         $this->configure($this->composer->getPackage()->getExtra(), 'composer-patches');
+        $this->lockFile = new JsonFile(
+            $this->getPatchesLockFilePath(),
+            null,
+            $this->io
+        );
+        $this->locker = new Locker($this->lockFile);
+    }
+
+    /**
+     * Get the path to the current patches lock file.
+     */
+    public function getPatchesLockFilePath(): string
+    {
+        $composer_file = \Composer\Factory::getComposerFile();
+
+        $dir = dirname(realpath($composer_file));
+        $base = pathinfo($composer_file, \PATHINFO_FILENAME);
+
+        $lock_file = $this->getConfig('patches-lock-file');
+        if (!is_string($lock_file)) {
+            $lock_file = 'patches-lock-file';
+        }
```

Comment:
This block isn't necessary anymore -- composer configurable plugin handles setting a default for you, so if it's not specified, it's whatever is hardcoded in the config above.